### PR TITLE
travis-ci: build with latest autotools

### LINKED ISSUE #925: flux-core fails to build for ppc64le in the buildfarm
```
Date: Tue, 13 Dec 2016 23:43:33 +0000
From: "Foraker, Jim" <foraker1@llnl.gov>
To: Jim Garlick <garlick@llnl.gov>
CC: "Mark A. Grondona" <mgrondona@llnl.gov>
Subject: flux-core fails to build for ppc64le in the buildfarm

     Sadly true.  Here’s a build log from trying to rebuild the current
ch6-chaotic version, 0.4.1-1.ch6:

http://tossbuild.llnl.gov/kojifiles/work/tasks/8401/28401/build.log

It looks like maybe architecture detection is going a little haywire.
Obviously not critical, but you’ll need to either fix it up or sprinkle
some ExclusiveArch magic into your spec file the next time you build on
ch6-chaotic.

     Jim
```

### PR DESCRIPTION
**Pull request by grondo**
Download and install latest autotools in travis builds.

This should resolve #925, though that is a bit difficult to test for certain...

Switching to spack for dependencies can come later, as discussed in #920

### CODE REVIEW

File: src/test/travis-dep-builder.sh [hunk 1]
```diff
@@ -16,7 +16,10 @@ https://github.com/zeromq/czmq/archive/v3.0.2.tar.gz \
 http://downloads.sourceforge.net/ltp/lcov-1.10.tar.gz \
 http://www.open-mpi.org/software/hwloc/v1.11/downloads/hwloc-1.11.0.tar.gz \
 http://www.digip.org/jansson/releases/jansson-2.9.tar.gz \
-http://www.mpich.org/static/downloads/3.1.4/mpich-3.1.4.tar.gz"
+http://www.mpich.org/static/downloads/3.1.4/mpich-3.1.4.tar.gz
```
Comment (chu11) [inline src/test/travis-dep-builder.sh]: should there be a "\\" contiunation at the end of this line?

Comment (grondo) [inline src/test/travis-dep-builder.sh]: Yes, I suppose so, thanks. Pushed a fix.

**Comment by garlick** [comment]
Is this one ready?

**Comment by grondo** [comment]
I think the spack solution is better, but there is still work to be done in
upstream spack, so I think this simple approach should work for 0.7.0
milestone.

**Comment by grondo** [comment]
I'll rebase this shortly once I get access to a computer

**Comment by grondo** [comment]
Ok, got around to a rebase on current master.

**Comment by garlick** [comment]
Great thanks!  Merging.

**Pull request merged**